File: genetic-packer/src/main/java/genetic/packer/evolution/fitness/FitnessComponents.java
package genetic.packer.evolution.fitness;

import genetic.api.builder.HasBuilder;
import javaslang.collection.Traversable;

/**
 * @author piotr.larysz
 */
public class FitnessComponents extends TranslatedFitness<FitnessComponents, Double> {

    private Traversable<Boolean> intersections;

    private Double volume;

    @HasBuilder
    public FitnessComponents(
        FitnessTranslator<FitnessComponents, Double> totalFitnessTranslator,
        FitnessTranslator<FitnessComponents, String> fitnessExplainedTranslator
    ) {
        super(totalFitnessTranslator, fitnessExplainedTranslator);
    }

    @Override
    protected FitnessComponents self() {
        return this;
    }

    public Traversable<Boolean> getIntersections() {
        return intersections;
    }

    public void setIntersections(Traversable<Boolean> intersections) {
        this.intersections = intersections;
    }

    public Double getVolume() {
        return volume;
    }

    public void setVolume(Double volume) {
        this.volume = volume;
    }
}

File: genetic-packer/src/main/java/genetic/packer/evolution/fitness/FitnessComponents.kt
package genetic.packer.evolution.fitness

import genetic.api.fitness.FitnessTranslator
import genetic.api.fitness.TranslatedFitness

class FitnessComponents(
    totalFitnessTranslator: FitnessTranslator<FitnessComponents, Double>,
    fitnessExplainedTranslator: FitnessTranslator<FitnessComponents, String>,
    val intersections: Sequence<Boolean>,
    val volume: Double
) : TranslatedFitness<FitnessComponents>(totalFitnessTranslator, fitnessExplainedTranslator) {

  override fun self() = this
}